### ISSUE TITLE
.drawio files not opening in draw.io

### ISSUE THREAD

**Issue by dougti** [pawelrojek/nextcloud-drawio#56]
Hi,
 I've added the MIME type to `/config/mimetypemapping.json`, as per the instructions, however files with  the _.drawio_ extension are still being sent to the browser for download. There is also no option to open the file with draw.io via the context menu.

I've also tried adding "_application/x-drawio_" to `/etc/apache2/mods-enabled/mime.conf` & reloading Apache, however it made no difference.

Have I missed something?

v0.9.1 (NC 13.06)

For full disclosure, there's also an external reverse proxy (nginx) that sits between the client & the internal webserver, however I don't think should make a difference. It's the internal webserver/Nextcloud which should be directing the client to draw.io...

Cheers

**Comment by dougti** [comment]
I've also run `occ maintenance:mimetype:update-js`, as per the [docs](https://docs.nextcloud.com/server/13/admin_manual/configuration_mimetypes/index.html), but still no luck.

**Comment by mistergibson** [comment]
I had something similar happen. I changed the file extension to .drawio and it BROKE association with the Draw.io tool. renaming it '.xml' as the extension didn't work either. Now it wants to download all my drawings instead of opening them up for editing. Me thinks some attention needs be paid to the file extension and the association with the tool. Just say'n. (Great tool btw, love it when it works).

**Comment by valkalon** [comment]
Same issue for me as @dougti . Using nextcloud 16.0.3
.xml works fine, .drawio does not

**Comment by topicomha** [comment]
I was able to get it to work using the instructions from the [App's github page](https://github.com/pawelrojek/nextcloud-drawio)

> To proceed, just copy /resources/config/mimetypemapping.dist.json to /config/mimetypemapping.json (in the config/ folder at Nextcloud’s root directory; the file should be stored next to the config.php file). Afterwards add the two following line just after the “_comment” lines.
> `"drawio": ["application/x-drawio"],`
>
> If all other mimetypes are not working properly, just run the following command:
> `sudo -u www-data php occ files:scan --all`
>

**Comment by GideonWolfe** [comment]
I'm experiencing an issue where `xml` files are no longer opened by drawio. I have been using `xml` files as Diagrams for months, but all of a sudden when I click it it gets sent to FF for download.

**Comment by topicomha** [comment]
You could check the settings and make sure the "Associate XML files with Draw.io?" setting is set to "Yes". You can find the setting in the "Additional Settings" (in the Administration Section) at the bottom of the left side menu on the "Settings" page.

![chrome_RK8gNPkNQ7](https://user-images.githubusercontent.com/10187806/69922833-74f14a00-1454-11ea-844f-8773d2383ee8.png)

**Comment by GideonWolfe** [comment]
@topicomha Yes this setting was set to "Yes". I think their might have been an issue with the mimetypemapping file when I edited it to enable Keepweb. I have since migrated to a new server and it's all working great.

**Comment by Axolord** [comment]
Just clicking the "save" button resolves the issue for me. I think the issue can be closed or does anyone experience any further problems?

**Comment by bitops** [comment]
> Just clicking the "save" button resolves the issue for me. I think the issue can be closed or does anyone experience any further problems?

I had the same experience - I simply had to visit the "Additional Settings" page on my NC instance and hit "Save" - once I did that the association for `.drawio` files began working again. 🥳

**Comment by aov-arnold** [comment]
Changing a setting and clicking the save button doesn't work for me on NC 24.0.4.1.

- When clicking a .drawio file in the files app it's opened with the drawio app correctly.
- When clicking on a share link for a shared .drawio file it only downloads the file.

@pawelrojek Can you reproduce the issue?

**Comment by ocervell** [comment]
Same issue on my side, since latest update clicking on drawio files just downloads them.

**Comment by adam-op** [comment]
Hi
NC 24.0.6 DrawIO Integration 1.3 Firefox Ubuntu Latest

For me also XML files are sometimes not opened but presented as Plaintext XML files
drawio files never opened yet
Does someone know a workaround if draw.io and XML all do not work to open in editing embedded mode?
Why is there no function in Nextcloud DrawIO Integration to right click and select Open/Edit in draw.io ?
